Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/profile_page.tsx

```diff
@@ -452,17 +459,139 @@ export class ProfilePage extends React.Component<Properties, State> {
       </div>);
   }
 
+  private onFirstNameChange(value: string) {
+    this.state.newIdentity.firstName = value;
+    if(this.props.identity.firstName === this.state.newIdentity.firstName) {
+      this.setState({
+        newIdentity: this.state.newIdentity,
+        isProfileChanged: false
```

Comment:
The way that ```isProfileChanged``` is handled is a bug. For example, change the first name from ```XXX``` to ```YYY```, then change last name from ```QQQ``` to ```ZZZ```, then change first name from ```YYY``` back to ```XXX```. ```isProfileChanged``` will be reset to false even though the last name has changed.

To fix this, once ```isProfileChanged``` is set to ```true```, never set it to ```false```. That will cover 90% of use cases and the small exception isn't too important.